web_client/utils/promise.jsx

```diff
@@ -1,4 +1,155 @@
 import { Promise } from 'bluebird';
+import scatter from './scatter';
 
 export { Promise };
+
+export const SCHEDULES = {
+  DYNAMIC: 'dynamic',
+  ROUND: 'round',
+  STATIC: 'static'
+};
+
+export const map = (args) => {
```

Comment:
Also, why not destructure `args` directly, including specifying the default values?